virtool/samples/data.py

```diff
@@ -284,6 +289,7 @@ async def create(
             user_id,
             JobRights(),
             0,
+            job_id,
```

Comment:
Gonna need @igboyes to chime in on this one, but I really am not sold on how this is currently structured as it is not atomic. It shouldn't happen much, but as written you could end up having a sample which reference a non-existent `job` if creating the job document fails. 

Unfortunately this is an artifact of how we designed the `dataLayer` since `dataLayer` methods are never passed a session. It could be fixed by moving most of the job creation logic into a `db` method, but that will essentially make the jobs `dataLayer.create` method redundant, as I suspect that method is only ever really called from other `dataLayers`. Also it means any `dataLayer` that spawns a job would have to get a copy of the `jobsClient`.

Reply:
Agreed. Please guide @SnickerDoodlePop on fixing this.

Focus only on samples for this PR. He can fix the other occurences in another PR + issue.